${parameter=word} not recognized as assignment

#### For bugs
- Rule Id (if any, e.g. SC1000): SC2223
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

`${parameter=word}` is very similar `${parameter:=word}`, except that it substitutes parameter even if it is defined *null*. Both are [specified by POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02). Shellcheck recognizes the latter as an assignment, but not the former and issues a confusing warning.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

: "${foo:=foo}"  # works
: "${bar=bar}"  # warns "SC2154: bar is referenced but not assigned."

echo "$foo $bar"
```

#### Here's what shellcheck currently says:

> SC2154: bar is referenced but not assigned.


#### Here's what I wanted or expected to see:

nothing

--- a/src/ShellCheck/Analytics.hs
+++ b/src/ShellCheck/Analytics.hs
@@ -2205,6 +2205,7 @@ prop_checkUnassignedReferences36= verifyNotTree checkUnassignedReferences "read 
 prop_checkUnassignedReferences37= verifyNotTree checkUnassignedReferences "var=howdy; printf -v 'array[0]' %s \"$var\"; printf %s \"${array[0]}\";"
 prop_checkUnassignedReferences38= verifyTree (checkUnassignedReferences' True) "echo $VAR"
 prop_checkUnassignedReferences39= verifyNotTree checkUnassignedReferences "builtin export var=4; echo $var"
+prop_checkUnassignedReferences40= verifyNotTree checkUnassignedReferences ": ${foo=bar}"
 
 checkUnassignedReferences = checkUnassignedReferences' False
 checkUnassignedReferences' includeGlobals params t = warnings

--- a/src/ShellCheck/AnalyzerLib.hs
+++ b/src/ShellCheck/AnalyzerLib.hs
@@ -508,7 +508,7 @@ getModifiedVariables t =
         T_DollarBraced _ _ l -> maybeToList $ do
             let string = bracedString t
             let modifier = getBracedModifier string
-            guard $ ":=" `isPrefixOf` modifier
+            guard $ any (`isPrefixOf` modifier) ["=", ":="]
             return (t, t, getBracedReference string, DataString $ SourceFrom [l])
 
         t@(T_FdRedirect _ ('{':var) op) -> -- {foo}>&2 modifies foo

